x-pack/plugins/reporting/common/types/index.ts

```diff
@@ -71,6 +71,7 @@ export interface ReportSource {
    */
   jobtype: string; // refers to `ExportTypeDefinition.jobType`
   created_by: string | false; // username or `false` if security is disabled. Used for ensuring users can only access the reports they've created.
+  error_code?: string;
```

Comment:
@jloleysens would it be possible to move this field under `output`?

The organization of fields as it was, used top-level fields for things that may be defined when the job is created, and `output` had containment of the fields that may only be defined after report job execution.

Reply:
Happy with this suggestion, I'll move it in a follow up PR